tests/cloudapp.js

```diff
@@ -35,78 +43,134 @@ describe('Cloud App', function() {
-
-    // TODO: get system and user space mongo pod name
-
-    // TODO: portforward both
-
-    // TODO: create connections to mongos
-    const userUrl = 'mongodb://admin:uD3DIIhmugC3eQbFCEaMP06DjqtHOiBB5al7Tltw@localhost:27017';
+    console.log('fhc init');
+    await fhc.init(config.host, config.user, config.pass);
+    console.log('creat project');
+    project = await fhc.createProject(`${config.prefix}${Date.now()}`, 'bare_project');
+
+    // create form and theme and associate it with the project
+    console.log('creat form and theme');
+    const formDef = await services.createForm(project.guid);
+    form = formDef.form;
+    theme = formDef.theme;
+
+    // cloud app
+    console.log('import cloud app');
+    const zipFile = path.resolve(__dirname, '../cloudapp.zip')
+    services.zipFolder(path.resolve(__dirname, '../cloud_app'), zipFile);
+    cloudApp = await fhc.importAppZip('Cloud App', project.guid, zipFile, 'cloud_nodejs');
+    console.log('deploy cloud app');
+    try {
+      await fhc.appDeploy(cloudApp.guid, config.env, config.runtime);
+    } catch (_) {
+      // issue with openshift - cloud app should be still deployed after some time
+      await new Promise(resolve => setTimeout(resolve, 120000));
+    }
+    console.log('wait for cloud app to be ready');
+    await new Promise(resolve => setTimeout(resolve, 60000));
+    console.log('create data with cloud app');
+    let cloudAppUrl = await fhc.getCloudUrl(cloudApp.guid, config.env);
+    await request({ uri: `${cloudAppUrl}/test`, json: true });
+
+    // client app
+    console.log('import client app');
+    const clientApp = await fhc.importApp(project.guid, 'Client App', 'client_advanced_hybrid', 'https://github.com/feedhenry-templates/helloworld-app.git', 'master', config.env);
+    const connections = await fhc.connectionsList(project.guid);
+    services.createFHConfigFile(clientApp, project, config.host, connections[0]);
+    console.log('run client app');
+    await services.runClientApp();
+
+    // form submission
+    console.log('create form submission');
+    cloudAppUrl = await fhc.getCloudUrl(cloudApp.guid, config.env);
+    var options = {
+      uri: `${cloudAppUrl}/submission`,
+      method: 'POST',
+      json: { form, theme, clientApp }
+    };
+    await request(options);
+
+    // portforward both
+    console.log('port-forward mongo');
+    const { userAdminPassword, systemAdminPassword } = await mongo.portForward();
+
+    // create connections to mongos
+    console.log('connect to mongo');
+    const userUrl = `mongodb://admin:${userAdminPassword}@localhost:27017`;
     userClient = await MongoClient.connect(userUrl);
+    const systemUrl = `mongodb://admin:${systemAdminPassword}@localhost:27018`;
+    systemClient = await MongoClient.connect(systemUrl);
 
-    // TODO: get cloud app pod name
-
+    // get cloud app id
+    cloudAppId = cloudAppUrl.substring(cloudAppUrl.indexOf('//') + 2).split('.')[0].split('-')[1];
```

Comment:
@jhellar can't be cloudAppId fetched from results of commands importAppZip or appDeploy?

Reply:
@flashlee I also tried to get this id with fhc, but didn't find it